include/solvers/navier_stokes_base.h

```diff
@@ -1266,11 +1268,18 @@ template <int dim, typename VectorType, typename DofsType>
 void
 NavierStokesBase<dim, VectorType, DofsType>::refine_mesh_kelly()
 {
+  if (dynamic_cast<parallel::distributed::Triangulation<dim> *>(
```

Comment:
Adaptive mesh refinement is unfortunately not supported at the moment for `p:f:t`. However, there is already some work in that direction (see https://github.com/dealii/dealii/pull/8834).